resources/newssite/news-nuxt/composables/provide-locale.js

```diff
@@ -0,0 +1,37 @@
+import { provide } from "vue";
+import { useHead } from "nuxt/app";
+import { content as contentEn } from "~/data/en/content";
+import { settings as settingsEn } from "~/data/en/dialog";
+import { footer as footerEn } from "~/data/en/footer";
+import * as buttonsEn from "~/data/en/buttons";
+import * as linksEn from "~/data/en/links";
+
+const strings = {
+    en: {
+        content: contentEn,
+        settings: settingsEn,
+        footer: footerEn,
+        buttons: buttonsEn,
+        links: linksEn,
+    },
+};
+
+export function provideLocale() {
+    const defaultLanguage = "en";
+    const urlParams = new URLSearchParams(window.location.search);
+    const dir = urlParams.get("dir") ?? "ltr";
+    const langFromUrl = urlParams.get("lang");
+    const lang = langFromUrl && langFromUrl in strings ? langFromUrl : defaultLanguage;
+
+    useHead({
```

Comment:
@danielroe  - saw a deprecation warning - are there other options to achieve the same thing?

Reply:
I think that might be about where you are importing it from.